toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/GeospatialTrackingCameraController.kt

```diff
@@ -62,10 +64,59 @@ internal class GeospatialTrackingCameraController(
         (context.getSystemService(Context.WINDOW_SERVICE) as? WindowManager)?.defaultDisplay
     }
 
+    private var lastEarthState: EarthState? = null
+
     override fun updateCamera(frame: Frame, session: Session) {
         session.earth?.let { earth ->
+            if (!hasSetOriginCamera && earth.earthState != lastEarthState) {
+                // This code will check the earth state for any initialization errors
+                // We don't want to call onError after the WorldScaleSceneView is initialized,
+                // so we do this before we set the origin camera the first time, which will trigger
+                // the change to Initialized. Therefore this code should only run while WorldScaleSceneView
+                // is initializing.
+                lastEarthState = earth.earthState
+                when (earth.earthState) {
+                    EarthState.ENABLED -> {}
+                    EarthState.ERROR_INTERNAL, EarthState.ERROR_GEOSPATIAL_MODE_DISABLED -> {
+                        onError(
+                            IllegalStateException(
+                                "WorldScaleSceneView has encountered an internal error. The app should not attempt to recover from this error. Please see the Android logs for additional information."
+                            )
+                        )
+                    }
+
+                    EarthState.ERROR_NOT_AUTHORIZED -> {
+                        onError(
+                            IllegalStateException(
```

Comment:
Per discussion, lets propagate a specific exception for this:
```suggestion
                            ArCoreAuthorizationException(
```